### PR TITLE
target/riscv: use buf_get_uXX instead of manual bit shift

### PR DESCRIPTION
**Pull request by fkhaidari**
Replaced manual bit shift with buf_get_u64/buf_get_u32

### CODE REVIEW

**Review by en-sc — DISMISSED** [review]
LGTM (reviewed internally).

**Comment by aap-sc** [comment]
@JanMatCodasip would you kindly take a look too?

File: src/target/riscv/riscv-013.c [hunk 1]
```diff
@@ -4486,27 +4480,21 @@ static int write_memory_bus_v1(struct target *target, target_addr_t address,
 
 			uint32_t sbvalue[4] = { 0 };
 			if (size > 12) {
-				sbvalue[3] = ((uint32_t)p[12]) |
-						(((uint32_t)p[13]) << 8) |
-						(((uint32_t)p[14]) << 16) |
-						(((uint32_t)p[15]) << 24);
+				sbvalue[3] = buf_get_u32(&p[12],
+						/* first = */ 0, /* bit_num = */ 32);
```
Comment (JanMatCodasip) [inline src/target/riscv/riscv-013.c:4534]: Pedantic detail: I'd add braces here to not make code readers wonder about C operator precedence.

```suggestion
				sbvalue[3] = buf_get_u32(&(p[12]),
						/* first = */ 0, /* bit_num = */ 32);
```

File: src/target/riscv/riscv-013.c [hunk 2]
```diff
@@ -4486,27 +4480,21 @@ static int write_memory_bus_v1(struct target *target, target_addr_t address,
...
 						RISCV_DELAY_BASE);
 			}
 
 			if (size > 8) {
-				sbvalue[2] = ((uint32_t)p[8]) |
-						(((uint32_t)p[9]) << 8) |
-						(((uint32_t)p[10]) << 16) |
-						(((uint32_t)p[11]) << 24);
+				sbvalue[2] = buf_get_u32(&p[8],
+						/* first = */ 0, /* bit_num = */ 32);
```
Comment (JanMatCodasip) [inline src/target/riscv/riscv-013.c:4541]:
```suggestion
				sbvalue[2] = buf_get_u32(&(p[8]),
						/* first = */ 0, /* bit_num = */ 32);
```

File: src/target/riscv/riscv-013.c [hunk 3]
```diff
@@ -4486,27 +4480,21 @@ static int write_memory_bus_v1(struct target *target, target_addr_t address,
...
 				riscv_batch_add_dm_write(batch, DM_SBDATA2, sbvalue[2], false,
 						RISCV_DELAY_BASE);
 			}
 			if (size > 4) {
-				sbvalue[1] = ((uint32_t)p[4]) |
-						(((uint32_t)p[5]) << 8) |
-						(((uint32_t)p[6]) << 16) |
-						(((uint32_t)p[7]) << 24);
+				sbvalue[1] = buf_get_u32(&p[4],
```
Comment (JanMatCodasip) [inline src/target/riscv/riscv-013.c:4546]:
```suggestion
				sbvalue[1] = buf_get_u32(&(p[4]),
```

**Review by JanMatCodasip — DISMISSED** [review]
LGTM, thanks.

File: src/target/riscv/riscv-013.c (hunk 1 above)
Comment (fk-sc) [reply src/target/riscv/riscv-013.c]: Addressed

File: src/target/riscv/riscv-013.c (hunk 2 above)
Comment (fk-sc) [reply src/target/riscv/riscv-013.c]: Addressed

File: src/target/riscv/riscv-013.c (hunk 3 above)
Comment (fkhaidari) [reply src/target/riscv/riscv-013.c:4546]: Looks like checkpatch doesn't allow this

Comment (JanMatCodasip) [reply src/target/riscv/riscv-013.c:4546]: Ok, no problem, thanks for trying it.

**Review by JanMatCodasip — DISMISSED** [review]

**Review by aap-sc — APPROVED** [review]

**Review by JanMatCodasip — APPROVED** [review]

**Pull request merged**